### ISSUE TITLE
Bug which causes fluctuations of great quantities of money

### ISSUE THREAD

**Issue by KeyboardMonkey95** [settintotrieste/Victoria-2-Cold-War-Enhancement-Mod-CWE#232]
This seems to be the same bug that i reported some time ago, but this time it didn't cause any immediate crashes as far as i can tell.
Basically, i started a new game with Italy with the 1.06 release as of 25th of June and around the middle of 1989 the balance starts to fluctuate on the order of +/- Millions a day. 

To give you an example it may go: one day +5M, the next -3M, and the one after that +400k and so on.
In less than a month it sent me to a budget of 150M and i think (i'm not sure if that was the cause) it bankrupted the PRC.

I was the 5th GP with a budget of ~20M, an Industrial Score of +3000, almost 60% literacy and all technologies available up to that point unlocked.

------------
_Previous bug was #151_ - ARR8

**Comment by settintotrieste** [comment]
Thanks for the bug report. May I have a copy of your savegame to analyse the cause of the problem?

**Comment by KeyboardMonkey95** [comment]
[FIGA.zip](https://github.com/settintotrieste/Victoria-2-Cold-War-Enhancement-Mod-CWE/files/1116973/FIGA.zip)

Here's the savegame, the bug happens in 1/2 months.

**Comment by settintotrieste** [comment]
Thank you!

**Comment by settintotrieste** [comment]
I have done some changes which has fixed the problem.

**Comment by settintotrieste** [comment]
Need more feedback to see if it is still occuring on the latest Github version.

**Comment by settintotrieste** [comment]
@ARR8 would it be possible if you could help me do a play through from the date in the savegame keyboard monkey provided to 2046 to see if the huge money fluctuations still occur? This is because I don't have enough time during the weekdays and this is a serious bug.

**Comment by KeyboardMonkey95** [comment]
I continued to play after the first fixes you made yesterday and the bug reappeared a year later, earlier today i resumed after the 'Major Bugfixes' commit and that seemed to solve it as well.

But i have the feeling the that bug is caused by an overflow, which would explain the big negative values, and that those fixes are just delaying whatever causes it.

**Comment by settintotrieste** [comment]
From my investigations, it is due to a bug causing the production of factories to suddenly become excessively huge like 1.79 billion which then causes the capitalists to earn $1.79 plus billion causing a stack overflow of the national bank and the upper strata tax income causing the government treasury to rapidly gain in funds. I suspect that the cause of this is due to a hard cap on how much total throughput % a factory can have and when it exceeds that amount, the throughput becomes negative triggering off the chain of events observed. The only know fix so far is to cap the total factory throughout level by reducing the bonuses modifiers give to throughput.

**Comment by settintotrieste** [comment]
If you can keep playing that save using the latest GitHub version and let me know the year when the bug returns.

**Comment by ARR8** [comment]
I'll try to take a look, as well.

**Comment by settintotrieste** [comment]
@ARR8 Thanks!

**Comment by ARR8** [comment]
CNR

**Comment by KeyboardMonkey95** [comment]
Update after further playing with that save file:

I formed the EU, integrated all of Europe (Russia Included), and kept playing until 2052 after unlocking all Industry techs (and almost all of the other groups) and the bug did not reappear. 

My Economy was obviously immense with a score of over 40.000 and a population of 260M pops, my budget was always in the tens of Millions, often above 150M, but as i said the bug did not appear nor did any similar ones.

I stopped there for two reasons:

1. My computer isn't very powerful and, because of that, the game became very slow, taking some seconds for every day to pass at speed 3.

2. I got bored because i had done all i wanted to do for the play-trough by uniting all of Europe.
